### PR TITLE
[PING] Do not abort (as OoM) when request size = 0

### PR DESCRIPTION
**Pull request by SergeGautherie**
`ping -l 0 reactos.org` is valid.

### CODE REVIEW

File: base/applications/network/ping/ping.c [hunk 1]
```diff
@@ -418,19 +418,26 @@ static
 void
 Ping(void)
 {
-    PVOID ReplyBuffer = NULL;
-    PVOID SendBuffer = NULL;
+    PVOID ReplyBuffer;
+    PVOID SendBuffer;
```
Comment (Extravert-ir) [inline base/applications/network/ping/ping.c]:
```suggestion
    PVOID SendBuffer = NULL;
```

File: base/applications/network/ping/ping.c [hunk 2]
```diff
@@ -418,19 +418,26 @@ static
...
 
-    SendBuffer = malloc(RequestSize);
-    if (SendBuffer == NULL)
+    if (RequestSize == 0)
     {
-        ConResPrintf(StdErr, IDS_NO_RESOURCES);
-        exit(1);
+        SendBuffer = NULL;
     }
+    else
```
Comment (Extravert-ir) [inline base/applications/network/ping/ping.c]:
```suggestion
    if (RequestSize > 0)
```

File: base/applications/network/ping/ping.c (hunk 1 above)
Comment (learn-more) [reply base/applications/network/ping/ping.c]: Why?

Comment (SergeGautherie) [reply base/applications/network/ping/ping.c]: Reverted.

File: base/applications/network/ping/ping.c (hunk 2 above)
Comment (SergeGautherie) [reply base/applications/network/ping/ping.c]: Modified.

**Pull request merged**